rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -513,7 +575,18 @@ void deserialize_array<std::string>(
 {
     (void)call_new;
     if (member->array_size_ && !member->is_upper_bound_) {
-        deser.deserializeArray(((rosidl_generator_c__String*)field)->data, member->array_size_);
+        auto deser_field = (rosidl_generator_c__String*)field;
+        std::string stringseq[member->array_size_];
+        deser.deserializeArray(stringseq, member->array_size_);
+        for (size_t i = 0; i < member->array_size_; ++i) {
+          // NOTE: Do we need to call init ?
```

Comment:
I don't think the conclusion is correct. The invocation of `rosidl_generator_c__String__fini` will read the `data` member and based on its content do conditional logic. But that variable has never been initialized. So I expect this to break in release code / corrupt memory. So imo the init call is required.